sdgym/benchmark.py

```diff
@@ -1051,6 +1054,46 @@ def _ensure_uniform_included(synthesizers):
         synthesizers.append(UniformSynthesizer)
 
 
+def _add_adjusted_scores(scores, timeout):
+    uniform_row = scores[scores['Synthesizer'] == 'UniformSynthesizer'].iloc[0]
+    uniform_fit_time = uniform_row['Train_Time']
+    uniform_sample_time = uniform_row['Sample_Time']
+    if 'Quality_Score' in scores.columns:
+        uniform_quality_score = uniform_row['Quality_Score']
+
+    def _get_adjusted_time(row):
+        if pd.isna(uniform_fit_time) or pd.isna(uniform_sample_time):
+            return
+        adjusted_total_time = uniform_fit_time
+        if 'error' in row.index and not pd.isna(row['error']):
+            if row['error'] == 'Synthesizer Timeout':
+                adjusted_total_time += timeout
+            else:
+                if not pd.isna(row['Train_Time']):
+                    adjusted_total_time += row['Train_Time']
+                if not pd.isna(row['Sample_Time']):
+                    adjusted_total_time += row['Sample_Time']
+            adjusted_total_time += uniform_sample_time
+        else:
+            adjusted_total_time += row['Train_Time']
+            adjusted_total_time += row['Sample_Time']
```

Comment:
if there is no error, do we still need to add the uniform_fit_time? Isn't this a case where no adjustment is needed?

Reply:
for graceful degradation, a UniformSynthesizer must always be fit, even if it's not used, so the fit time is included even if it wasn't needed

Reply:
ah I see